setup.py

```diff
@@ -34,23 +34,51 @@
 """
 
 
+def _get_freetype_config():
+    print("Trying freetype-config to find freetype library...")
+    try:
+        # pointer to freetype build directory (tweak as necessary)
+        return subprocess.check_output(
+            ['freetype-config', '--prefix']).strip().replace(
+            b'"', b'').decode()
+    except (OSError, subprocess.CalledProcessError):
+        return None
+
 def _get_freetype_with_ctypes():
     print("Using ctypes to find freetype library...")
     from ctypes.util import find_library
     ft_lib_path = find_library('freetype')
     if ft_lib_path is None:
         return None
     ft_lib_path = os.path.dirname(ft_lib_path)
+    if not sys.platform.startswith('linux') and \
+            not os.path.isfile(ft_lib_path):
+        return None
+    elif not os.path.isfile(ft_lib_path):
+        # try prefix since find_library doesn't give a full path on linux
```

Comment:
This doesn't work because of the  `os.path.dirname(ft_lib_path)` line above which returns an empty string. But the lines below manage to return a path (`/tmp/simon` or `/home/simon`), so the pkgconfig function is skipped:
```
❯ python setup.py develop
Trying freetype-config to find freetype library...
Using ctypes to find freetype library...
[1] > /home/simon/dev/aggdraw/setup.py(54)_get_freetype_with_ctypes()
-> ft_lib_path = os.path.dirname(ft_lib_path)
(Pdb++) ft_lib_path
'libfreetype.so.6'
(Pdb++) os.path.dirname(ft_lib_path)
''
(Pdb++) unt 65
[1] > /home/simon/dev/aggdraw/setup.py(65)_get_freetype_with_ctypes()
-> lib_path = os.path.realpath(os.path.join(ft_lib_path, '..'))
(Pdb++) ft_lib_path
''
(Pdb++) os.path.realpath(os.path.join(ft_lib_path, '..'))
'/home/simon/dev'
```